src/main/java/hng_java_boilerplate/waitlist/controller/WaitlistController.java

```diff
@@ -0,0 +1,40 @@
+package hng_java_boilerplate.waitlist.controller;
+
+import hng_java_boilerplate.email.EmailServices.EmailProducerService;
+import hng_java_boilerplate.waitlist.entity.Waitlist;
+import hng_java_boilerplate.waitlist.service.EmailService;
+import hng_java_boilerplate.waitlist.service.WaitlistService;
+import jakarta.validation.Valid;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+import java.util.HashMap;
+import java.util.Map;
+
+@RestController
+@RequestMapping("api/v1/waitlist")
+public class WaitlistController {
+    @Autowired
+    private WaitlistService waitlistService;
+    @Autowired
+    private EmailProducerService emailProducerService;
```

Comment:
Make this consistent with your other fields. Prefer constructor injection.

Reply:
Done. Thanks